super-ddf.c

```diff
@@ -3865,6 +3871,7 @@ static int validate_geometry_ddf(struct supertype *st,
 		}
 		close(cfd);
```

Comment:
there's still a stupid `sra` leak here, if `load_super_ddf_all` returns nonzero.  Maybe just call `sysfs_free` before calling `load_super_ddf_all`?

Reply:
okay, doh. I had it there before. Doing it.